lib/puppetx/augeasproviders_grub/util.rb

```diff
@@ -122,40 +122,52 @@ def self.munge_grubby_value(value, flavor, grubby_info)
         return value
       end
 
-      # Return the location of the GRUB2 configuration on the system.
+      # Return the name of the current operating system or an empty string if
+      # not found.
+      #
+      # @return (String) The current operating system name
+      def self.os_name
+        # The usual fact
+        (Facter.value(:os) && Facter.value(:os)['name']) ||
+          # Legacy support
+          Facter.value(:operatingsystem) ||
+          # Fallback
+          ''
+      end
+
+      # Return the location of all valid GRUB2 configurations on the system.
       # Raise an error if not found.
       #
-      # @return (String) The full path to the GRUB2 configuration file.
-      def self.grub2_cfg_path
+      # @return (Array[String]) Paths to all system GRUB2 configuration files.
+      def self.grub2_cfg_paths
         paths = [
           '/etc/grub2.cfg',
-          '/boot/grub/grub.cfg',
-          '/boot/grub2/grub.cfg'
+          '/etc/grub2-efi.cfg',
+          "/boot/efi/EFI/#{self.os_name.downcase}/grub.cfg",
+          '/boot/grub2/grub.cfg',
+          '/boot/grub/grub.cfg'
         ]
 
-        if File.exist?('/sys/firmware/efi')
-          os_info = Facter.value(:os)
-          if os_info
-            os_name = Facter.value(:os)['name']
-          else
-            # Support for old versions of Facter
-            unless os_name
-              os_name = Facter.value(:operatingsystem)
-            end
+        paths.map do |path|
+          begin
+            File.realpath(path)
```

Comment:
The previous code also validated that the path was readable and not a directory.  If we wanted to keep this extra validation, we could do:

```suggestion
            real_path = File.realpath(path)
            real_path if (File.readable?(real_path) && !File.directory?(real_path))
```